src/components/Personalization/actions/setImageSource.js

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { showElements } from "../flicker";
+
+const IMAGE_TAG = "IMG";
+const SRC = "src";
+
+const isImage = element => element.tagName === IMAGE_TAG;
+const loadImage = url => {
+  const image = document.createElement(IMAGE_TAG);
+  image.src = url;
+};
+
+export default collect => {
+  return (settings, event) => {
+    const { elements, prehidingSelector } = event;
+    const { content: url, meta } = settings;
+
+    elements.filter(isImage).forEach(element => {
+      // Start downloading the image
+      loadImage(url);
+
+      // Remove "src" so there is no flicker
+      element.removeAttribute(SRC);
+
+      // Replace the image "src"
+      element.setAttribute(SRC, url);
```

Comment:
Yep, we have found it empirically, customers were complaining that they see flicker when swapping images. Apparently browsers have some internal mechanism that preserve the old image for a bit, when changing SRC. So the only way to avoid this to:
1. Pre-load the new image
2. Remove the old SRC
3. Add the new SRC